[16.0][ADD] quality_control_product_manufacturer: Added new module

anusriNPS opened the pull request:
1. Added a field under "quality_control_oca" module in "qc.inspection" model to display product image with respect to product_id available as part of selection made in reference field of the model.

2. Added new module which bridges "quality_control_oca" and "product_manufacturer" in order to display details about Manufacturer and Manufacturer Code related to product_id available as part of selection made in Reference field in the Inspection form.

SirPyTech commented on quality_control_oca/views/qc_inspection_view.xml:
```diff
@@ -184,39 +191,46 @@
-                                    /></span>
+                                <span class="float-end text-end">
+                                    <t t-esc="record.qty.value" />
+                                </span>
```
**suggestion**: There is a lot of refactoring in this commit: if it is needed, please move it to a dedicated `[REF]` commit so that it's more clear which changes are needed for the improvement that is being added.
Adding the `name` attribute to the pages can be considered a refactoring too.

anusriNPS replied:
Removed changes which were resulted as part of formatting XML. Keeping only changes relevant to requirement and hence used [IMP] tag in the commit for "quality_control_oca" module.

SirPyTech replied:
:+1:

SirPyTech commented on quality_control_product_manufacturer_oca/__manifest__.py:
```diff
@@ -0,0 +1,15 @@
+{
+    "name": "Quality Control Product Manufacturer OCA",
+    "version": "16.0.0.0.1",
```
**chore**: The first version number should be `16.0.1.0.0`, see https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#12version-numbers:
> The version number in the module manifest should be the Odoo major version (e.g. 12.0) followed by the module x.y.z version numbers. For example: 12.0.1.0.0 is expected for the first release of an 12.0 module.

anusriNPS replied:
Updated version as required

SirPyTech replied:
:+1:

SirPyTech commented on quality_control_product_manufacturer_oca/models/qc_inspection.py:
```diff
@@ -0,0 +1,13 @@
+
+class QcInspection(models.Model):
+    _inherit = "qc.inspection"
+    _description = "Quality control inspection with Manufacturer details"
```
**issue**: This is would overwrite the description of the model `qc.inspection`.
If you want to keep this note, please add it as a class docstring or a simple comment.

anusriNPS replied:
Updated using comment in the model file introduced

SirPyTech replied:
:+1:

SirPyTech commented on quality_control_product_manufacturer_oca/views/qc_inspection_views.xml:
```diff
@@ -0,0 +1,32 @@
+            <xpath expr="//page[@name='notes']" position="after">
+                <page
+                    string="Manufacturer"
+                    attrs="{'invisible': [('object_id', '=', False), ('object_id', '!=', 'product.product')]}"
```
**suggestion**: Since the fields in this page depend on the selected product, it would make sense to show it only if there is a product (`product_id`) selected, what do you think?

**question**: I'm not sure the field `object_id` could ever be `product.product`: because it is a `Reference` field and they are stored as `model.name,model_id`, could you please check if this condition is working as expected?

anusriNPS replied:
Updated domain using "product_id" as well. 

Please find answer in below comment related to "object_id" query

SirPyTech replied:
:+1:

SirPyTech commented on quality_control_product_manufacturer_oca/views/qc_inspection_views.xml:
```diff
@@ -0,0 +1,32 @@
+        <field name="arch" type="xml">
+            <xpath expr="//page[@name='notes']" position="after">
+                <page
+                    string="Manufacturer"
```
**suggestion**: Add a `name` attribute: it is useful for inheriting views (as you have just found out in this exact PR :wink:)

anusriNPS replied:
Yes, it was missed and I have updated now

SirPyTech replied:
:+1:

SirPyTech commented on quality_control_product_manufacturer_oca/views/qc_inspection_views.xml:
```diff
@@ -0,0 +1,32 @@
+        <field name="inherit_id" ref="quality_control_oca.qc_inspection_search_view" />
+        <field name="arch" type="xml">
+           <field name="test" position="after">
+                 <field name="manufacturer" />
+                 <field name="manufacturer_code" />
```
**question**: I believe that fields should be `store`d in order to be searchable, could you please check that searching using these fields is working as expected?
I tried in runboat but I can't select products for a related test: 
![image](https://github.com/user-attachments/assets/a7b01917-fa04-4be4-a0b2-93523eb65230)

anusriNPS replied:
Apologies for misleading information as part of commit message, which lead you to check under "Test" form for "Reference" field. So, I have updated as required.

The model "qc.inspection" has form view under "Inspection" where object_id can be found with "product.product". Runboat installation as below:

![image](https://github.com/user-attachments/assets/774a7202-24e6-4701-b423-aa2a8fc468c7)

Tree view:

![image](https://github.com/user-attachments/assets/9b1cd312-e99d-446c-8a29-f88fab842d5c)

Also, search is achievable without using store attribute for fields. Please find image as below where test1 has "manufacturer_code" as 1234556 and test3 has "manufacturer_code" as "123456" and result is as expected.

![image](https://github.com/user-attachments/assets/fc340355-8119-4b4a-8a50-2b5657d7c498)

SirPyTech replied:
Thanks for the explanation! It looks fine then :+1:

SirPyTech commented on quality_control_product_manufacturer_oca/__manifest__.py:
```diff
@@ -0,0 +1,15 @@
+{
+    "name": "Quality Control Product Manufacturer OCA",
```
**note**: The `_oca` suffix in the module's folder can be removed: it is normally used to avoid name clashing with Odoo's modules.
For instance, the module `quality_control_oca` has the `_oca` suffix so it doesn't clash with Odoo enterprise's module `quality_control`, see https://github.com/OCA/manufacture/pull/481#issuecomment-730439422.

anusriNPS replied:
Thank for the detailed reference. Removed "_oca" prefix in the newly introduced module

SirPyTech replied:
:+1:

SirPyTech commented on quality_control_oca/models/qc_inspection.py:
**chore**: The commit's title should start with `[IMP]` instead of `[ADD]`, see https://www.odoo.com/documentation/16.0/contributing/development/git_guidelines.html#tag-and-module-name:
> - [ADD] for adding new modules;
> - [IMP] for improvements: most of the changes done in development version are incremental improvements not related to another tag;
> [...]

In OCA we also have `[MIG]` (see https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#71commit-message).

anusriNPS replied:
[ADD] tag was wrongly used though I was aware of OCA convention for commit message. Will be cautious here on while making a commit message.

SirPyTech replied:
:+1:

SirPyTech reviewed (changes requested):
Thanks for the PR!

anusriNPS committed:
[IMP] quality_control_oca: Added field under qc_inspection

  Added "product_image" field to display product image related
to the product selected under "product_id" in the Inspection.

SirPyTech reviewed (approved):
Looks great :clap: thanks!

HekkiMelody commented on quality_control_oca/models/qc_inspection.py:
```diff
@@ -66,6 +66,9 @@ def _compute_product_id(self):
         help="Product associated with the inspection",
     )
+    product_image = fields.Image(
+        related="product_id.image_1920", string="Product Image", store=True
```
**question** (non-blocking): is it necessary to store this field? This would store a copy of the image on every inspection, instead of fetching it from the product itself.

anusriNPS replied:
I think its not necessary to have store for this field, will verify once and update it accordingly.

HekkiMelody reviewed (approved):
Code review, LGTM!

anusriNPS committed:
[ADD] quality_control_product_manufacturer: Added new module

  Added module "quality_control_product_manufacturer" to display
manufacturer details with respect to "product_id" available from
option selected from field "Reference" in the Inspection.

SirPyTech commented:
@OCA/manufacturing-maintainers this new Beta module already has a 2 approvals, that should be enough for merge, can someone have a look? Thanks!

ivantodorovich commented:
/ocabot merge minor

HekkiMelody commented:
Thank you @ivantodorovich !
